operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/KubernetesDependentResourceController.java

```diff
@@ -46,45 +43,16 @@ public KubernetesDependentResourceController(DependentResource<R, P> delegate,
             configuration.getDependentResourceClass());
   }
 
-  @SuppressWarnings("unchecked")
-  @Override
-  protected Persister<R, P> initPersister(DependentResource<R, P> delegate) {
-    return (delegate instanceof Persister) ? (Persister<R, P>) delegate : this;
-  }
-
-  @Override
-  public String descriptionFor(R resource) {
-    return String.format("'%s' %s dependent in namespace %s", resource.getMetadata().getName(),
-        resource.getFullResourceName(),
-        resource.getMetadata().getNamespace());
-  }
-
-  @Override
-  public EventSource initEventSource(EventSourceContext<P> context) {
-    this.client = context.getClient();
-    informer = new InformerEventSource<>(configuration, context);
-    return informer;
-  }
-
   @Override
-  public void createOrReplace(R dependentResource, Context context) {
-    client.resource(dependentResource).createOrReplace();
+  public Optional<EventSource> eventSource(EventSourceContext<P> context) {
+    var informer = new InformerEventSource<>(configuration, context);
+    ((KubernetesDependentResource) delegate).withInformerEventSource(informer);
+    return this.eventSource(context);
   }
 
   @Override
-  public R getFor(P primary, Context context) {
-    return informer.getAssociated(primary).orElse(null);
+  public Optional<R> getResource(P primaryResource) {
+    return delegate.getResource(primaryResource);
   }
 
-  public boolean owned() {
-    return getConfiguration().isOwned();
-  }
-
-  @Override
-  protected void createOrReplaceDependent(P primary, R dependent, Context context) {
```

Comment:
This feature needs to be restored for automatic owner reference management. This is something that people have requested.

Reply:
Will add this logic to the `KubernetesDependentResource` , but good point.